backend/src/question/question.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from "@nestjs/common";
+import { Category, Question, Topic } from "@prisma/client";
+
+import { PrismaService } from "src/core/prisma.service";
+
+import { DifficultyLevel } from "~shared/types/base";
+
+@Injectable()
+export class QuestionService {
+  constructor(private prisma: PrismaService) {}
+
+  async getQuestionById(id: number): Promise<Question | null> {
+    return this.prisma.question.findUnique({
+      where: { id: id },
+    });
+  }
```

Comment:
We can actually ask Prisma to include the topics and categories during this query, and then flatten the structure at the controller layer. The return type of this function will be a union of a few objects, but we can leave TypeScript to handle the inference of types.

Doing it this way will mean we can get rid of the other 2 methods and also reduce the amount of business logic in the controller.

```suggestion
  async getQuestionById(id: number) {
    return this.prisma.question.findUnique({
      where: { id: id },
      include: {
        topics: true,
        category: true,
      },
    });
  }
```

The data transformation in the controller can then be done with:
```
    const questionReturnType: GetQuestionRes = {
      title: question.title,
      category: question.category.title,
      topics: question.topics.flatMap((topic) => topic.name),
      description: question.description,
      hints: question.hints,
    };
```

Reply:
I tried using this but got an error that says that "Property 'category' does not exist on type 'Question'. Did you mean 'categoryId'?"